### PR TITLE
DictionarySlim trim

### PR DESCRIPTION
**Pull request by danmosemsft**
Relates to https://github.com/dotnet/corefxlab/issues/2592

2 commits - 
Remove Key/Value Enumerables to reduce instantiated size of type as discussed above
Remove Capacity to reduce unnecessarily publicness

### CODE REVIEW

**Comment by danmosemsft** [comment]
Oh, let me add XML comments too

File: tests/Microsoft.Experimental.Collections.Tests/DictionarySlimTests.cs [hunk 1]
```diff
@@ -723,6 +537,22 @@ public void UsedIEquatable2()
...
+    {
+        // Capacity is not exposed publicly, but is valuable in tests to help
+        // ensure everything is working as expected internally
+        public static int GetCapacity<TKey, TValue>(this DictionarySlim<TKey, TValue> dict) where TKey : IEquatable<TKey>
+        {
+            FieldInfo fi = typeof(DictionarySlim<TKey, TValue>).GetField("_entries", BindingFlags.NonPublic | BindingFlags.Instance);
+            Object entries = fi.GetValue(dict);
+
+            PropertyInfo pi = typeof(Array).GetProperty("Length");
+            int capacity = (int)pi.GetValue(entries);
```
Comment (safern) [inline tests/Microsoft.Experimental.Collections.Tests/DictionarySlimTests.cs]: We could just return this?

**Comment by danmosemsft** [comment]
OK now also 
Added XML docs (file temporarily included in the commit as I cannot find a way to consume $(OutputPath) in the project)
Removed Copy to (low value, extra size)

**Comment by TylerBrinkley** [comment]
I think a key and value collection are still both needed just not the nested Enumerator struct. I think that's what @jkotas meant.

**Comment by danmosemsft** [comment]
> I think a key and value collection are still both needed just not the nested Enumerator struct. I think that's what @jkotas meant.

I'm missing something, if we still have `ICollection<TKey>` but no Enumerator struct what would it return from `GetEnumerator<TKey>` ?

**Comment by TylerBrinkley** [comment]
yield return? Though that would create an iterator class. I guess we can just use a LINQ Select clause to replicate the Key and Value collections where needed.

**Comment by TylerBrinkley** [comment]
Looks good to me.

**Comment by danmosemsft** [comment]
> I guess we can just use a LINQ Select clause to replicate the Key and Value collections where needed.

Yes, that's what I did in the tests. As with anything - it is always easier to add than remove API.

**Comment by danmosemsft** [comment]
@safern is going to get the package signed and on NuGet in the next few days. After that is up @AnthonyLloyd are you interested in sending in the updated benchmark game submission? I think you deserve most of the credit.

**Comment by AnthonyLloyd** [comment]
> @safern is going to get the package signed and on NuGet in the next few days. After that is up @AnthonyLloyd are you interested in sending in the updated benchmark game submission? I think you deserve most of the credit.

Yes sure, thanks you all. It may be a good idea to wait until the package has a few downloads. In the benchmarks game they want to see that it's not just a package put up for the benchmark only.

**Comment by danmosemsft** [comment]
@anthonylloyd it already has 50K downloads :https://www.nuget.org/packages/Microsoft.Experimental.Collections/1.0.3-alpha

**Pull request merged**